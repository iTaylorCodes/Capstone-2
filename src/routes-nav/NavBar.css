.NavBar {
  background-color: #9dc183;
  display: flex;
  justify-content: space-between;
  padding: 1rem;
  border-bottom: 2px solid rgba(0, 0, 0, 0.125);
}

.NavBar a {
  text-shadow: 1px 1px black;
  color: white;
  font-size: large;
  font-weight: 500;
  text-decoration: none;
}

.NavBar-right a {
  margin-left: 1rem;
}

.NavBar .activeLink {
  text-shadow: 1px 1px black;
  color: white;
  font-size: large;
  font-weight: 800;
  text-decoration: none;
  border-bottom-width: 3px;
  border-bottom-style: solid;
  padding-bottom: 3px;
}
