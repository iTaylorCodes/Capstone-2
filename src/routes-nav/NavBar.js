import { useState } from "react";
import { NavLink, Link } from "react-router-dom";
import "./NavBar.css";

const NavBar = () => {
  // Handles setting currentUser until userContext is setup
  const [currentUser, setCurrentUser] = useState(false);
  return (
    <nav className="NavBar">
      <div className="NavBar-left">
        <NavLink
          className={(navData) => (navData.isActive ? "activeLink" : "")}
          to="/"
        >
          Neighborhood
        </NavLink>
      </div>
      {!currentUser ? (
        <div className="NavBar-right">
          <NavLink
            className={(navData) => (navData.isActive ? "activeLink" : "")}
            to="/signup"
          >
            Signup
          </NavLink>
          <NavLink
            className={(navData) => (navData.isActive ? "activeLink" : "")}
            to="/login"
            onClick={() => setCurrentUser(true)}
          >
            Login
          </NavLink>
        </div>
      ) : (
        <div className="NavBar-right">
          <NavLink
            className={(navData) => (navData.isActive ? "activeLink" : "")}
            to="/profile"
          >
            Profile
          </NavLink>
          <NavLink
            className={(navData) => (navData.isActive ? "activeLink" : "")}
            to="/favorites"
          >
            Favorites
          </NavLink>
          <Link to="/" onClick={() => setCurrentUser(false)}>
            Logout {currentUser.username}
          </Link>
        </div>
      )}
    </nav>
  );
};

export default NavBar;
