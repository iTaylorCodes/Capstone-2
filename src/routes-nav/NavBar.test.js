import { render, fireEvent } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import NavBar from "./NavBar";

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <NavBar />
    </MemoryRouter>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <NavBar />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("renders without crashing after login", function () {
  const { container, getByText } = render(
    <MemoryRouter>
      <NavBar />
    </MemoryRouter>
  );

  // Tests post login event
  const button = getByText("Login");
  fireEvent.click(button);
  expect(
    render(
      <MemoryRouter>
        <NavBar />
      </MemoryRouter>
    )
  ).toBeTruthy();
});
