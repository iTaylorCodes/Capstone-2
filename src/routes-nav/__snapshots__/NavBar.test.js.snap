// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="NavBar"
  >
    <div
      class="NavBar-left"
    >
      <a
        aria-current="page"
        class="activeLink"
        href="/"
      >
        Neighborhood
      </a>
    </div>
    <div
      class="NavBar-right"
    >
      <a
        class=""
        href="/signup"
      >
        Signup
      </a>
      <a
        class=""
        href="/login"
      >
        Login
      </a>
    </div>
  </nav>
</DocumentFragment>
`;
